### ISSUE TITLE
Namespace restricted users with federation not working

### ISSUE THREAD

**Issue by michbeck100** [neuvector/neuvector#1064]
**Environment**
Platform: Rancher
Platform Version: 2.7.6

**Describe the bug**
I have setup an instance of neuvector, with a single manager and multiple controller instances as described here: https://open-docs.neuvector.com/navigation/multicluster.
Additionally i have to restrict my OIDC users to certain namespaces with the role reader (based on https://open-docs.neuvector.com/configuration/users#namespace-restricted-users). Once a user logs in using OIDC, he will have only the role reader and can't navigate the different clusters. If I set the global role in the group role map to fedReader, the user has too many rights and can see all namespaces again.
A solution could be to set the namespace role to fedReader. But this is not allowed at the moment.

**To Reproduce**
1. Setup multi-cluster environment as descibed in https://open-docs.neuvector.com/navigation/multicluster
2. Generate OIDC config with group role map to restrict user to namespace
3. Login as OIDC user
4. Can't navigate between different clusters.
5. Change the global role to `fedReader` in the group role map
6. Login as OIDC user again
7. Can navigate between clusters, but can see all namespaces, not only the restricted namespaces

**Expected behavior**
Users should be able to navigate to all clusters and still be restricted to their namespaces

**Comment by michbeck100** [comment]
Any comments?

**Comment by williamlin-suse** [comment]
It's not supported to assign fedAdmin/fedReader role to namespaced users

**Comment by michbeck100** [comment]
Any news on this?

**Comment by gkosaka** [comment]
Yes this is a feature request in our system and on our roadmap. However, it is not committed to a specific release at this time.

**Comment by pna-nca** [comment]
I would also be interested in group - role assignments which support cluster name specification, not only namespace.
Would be helpful to have a workaround at least. :)

**Comment by becitsthere** [comment]
What is desired behavior of this user? A user can manage specific namespaces across all clusters? What is the use case for it?

**Comment by michbeck100** [comment]
The user would be limited to certain namespaces across all clusters, yes. Since we don't want to have a manager UI for every cluster, the federation feature is nice. 
But not everybody is allowed to view every namespace. There are even some namespaces that exist in multiple clusters, but even if not, it's better than having multiple manager UIs with multiple hostnames with multiple logins. 
I guess the main use case would be for read only access, not for admins.

**Comment by pna-nca** [comment]
> What is desired behavior of this user? A user can manage specific namespaces across all clusters? What is the use case for it?

I see two desired behaviors, one is a sort of workaround which is sufficient, probably easier to implement, but not great. And another one which is, let's say, complete.

The "workaround" behavior:
- A role can be defined which has permissions to access all clusters.
- The remaining parameters/assignments are not changed.
- When a user with a role to read all cluster accesses a resource, "namespace" scoping (if defined) is applied as it is now.
- The only difference (from the UI perspective) is that in multicluster / federated mode, a user can switch clusters in the top menu.

The "complete" behavior:
- A role can be defined which has a permission to access a defined set of cluster.
- By default a user has access to its "origin" cluster, so nothing changes from UI/internals perspective.
- However, one can specify which clusters a user has access to, similarly to namespaces we define.
- In simplified implementation, the granted namespaces access applies to all downstream clusters.
- In advanced implementation, the system has to create a role (set of roles) per cluster. Like the same set of permissions for a role, but this role is applicable only to a certain cluster.

**Comment by gkosaka** [comment]
Thanks for use case descriptions. We've logged them into our internal feature tracking system. We hope to improve federated role behaviors by end of this year.

**Comment by richardvdveen** [comment]
Based on https://github.com/neuvector/neuvector/issues/2064#issuecomment-3178111458 I was happy to read that wildcarded namespaces can be assigned to specific GroupRoleBindings. This issue seems to be the next blocker in my implementation, as I also have a federated cluster setup where I would like to give teams the fedReader role in their own namespaces across clusters.

Is there any update on this issue? Is there any way for me to view the roadmap? Thanks in advance!
